client-core/src/wallet/syncer_logic.rs

```diff
@@ -127,6 +129,44 @@ fn on_transaction_change(
     }
 
     memento.add_transaction_change(transaction_change);
+    Ok(())
+}
+
+/// update WalletStateMemento when import a transaction
+pub(crate) fn import_transaction(
```

Comment:
Maybe we should rename `syncer_logic` to `wallet_logic` or something when it's function goes beyond syncing, to prevent future confusion.